yal/assessments.py

```diff
@@ -150,3 +154,42 @@ async def state_survey_process_answer(self):
                 self.save_metadata("assessment_score", score)
 
         return await self.go_to_state("state_survey_question")
+
+    async def state_assessment_later_submit(self):
+        msisdn = normalise_phonenumber(self.inbound.from_addr)
+        whatsapp_id = msisdn.lstrip(" + ")
+
+        reminder_time = get_current_datetime() + timedelta(hours=23)
+        self.save_metadata("assessment_reminder", reminder_time.isoformat())
+        assessment_name = self.user.metadata.get(
+            "assessment_name", "sexual_health_literacy"
+        )
+
+        data = {
+            "assessment_reminder": reminder_time.isoformat(),
+            "assessment_name": assessment_name,
+        }
+
+        await rapidpro.update_profile(whatsapp_id, data, self.user.metadata)
+        return await self.go_to_state("state_assessment_later")
+
+    async def state_assessment_later(self):
+        question = self._(
+            "\n".join(
+                [
+                    "[persona_emoji] No worries, we get it!",
+                    "",
+                    "I'll send you a reminder message tomorrow, so you can come back "
+                    "and continue with these questions, then.",
+                    "",
+                    "Check you later 🤙🏾",
+                ]
+            )
+        )
+        return WhatsAppButtonState(
+            self,
+            question=question,
+            choices=[Choice("menu", "Go to main menu")],
+            error=get_generic_error(),
+            next=MainMenuApplication.START_STATE,
+        )
```

Comment:
I'm not sure about these states existing in `assessments`. They don't seem to interact with the rest of the generic assessment states. I think `onboarding` is probably a better place for this? Since it's specific to that flow?

Reply:
I do agree, but, all the flows follow the same "I'll do this later" flow where they set a reminder and ask the user if they'd like to return to the main menu 
https://miro.com/app/board/uXjVP-lrlK0=/?moveToWidget=3458764541062653134&cot=14